### PR TITLE
owner(ticdc): decouple changefeed from reactor

### LINKED ISSUE #9784: decouple changefeed from reactor
decouple changefeed from reactor

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #9784

### What is changed and how it works?
decouple changefeed from reactor

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -47,6 +46,21 @@ import (
 	"go.uber.org/zap"
 )
 
+// Changefeed is the tick logic of changefeed.
+type Changefeed interface {
+	// Tick is called periodically to drive the Processor's internal logic.
+	// The main logic of processor is in this function, including the calculation of many kinds of ts,
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:52]: Should processor be changefeed?

File: cdc/owner/changefeed.go [hunk 2]
```diff
@@ -143,21 +155,29 @@ type changefeed struct {
 	) (observer.Observer, error)
 
 	lastDDLTs uint64 // Timestamp of the last executed DDL. Only used for tests.
+
+	// The latest changefeed info and status from meta storage. they are updated in every Tick.
+	latestInfo   *model.ChangeFeedInfo
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:160]: Will it be better to put them into one struct?

File: cdc/owner/changefeed.go [hunk 3]
```diff
@@ -306,48 +336,48 @@ func (c *changefeed) checkStaleCheckpointTs(ctx cdcContext.Context, checkpointTs
 	return nil
 }
 
-func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*model.CaptureInfo) error {
-	adminJobPending := c.feedStateManager.Tick(c.state, c.resolvedTs)
-	preCheckpointTs := c.state.Info.GetCheckpointTs(c.state.Status)
+func (c *changefeed) tick(ctx cdcContext.Context,
+	cfInfo *model.ChangeFeedInfo,
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:340]: Seems that `cfStatus` and `cfInfo` are already updated into the changefeed sturct in the `Tick` method. So there is no need to pass them into `tick` method explicitly.

File: cdc/owner/changefeed.go [hunk 4]
```diff
@@ -306,48 +336,48 @@ func (c *changefeed) checkStaleCheckpointTs(ctx cdcContext.Context, checkpointTs
 	return nil
 }
 
-func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*model.CaptureInfo) error {
-	adminJobPending := c.feedStateManager.Tick(c.state, c.resolvedTs)
-	preCheckpointTs := c.state.Info.GetCheckpointTs(c.state.Status)
+func (c *changefeed) tick(ctx cdcContext.Context,
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:339]: Please add a comment to explain the meaning of this method's return value.

File: cdc/owner/changefeed.go [hunk 5]
```diff
@@ -399,30 +429,32 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
...
 }
 
-func (c *changefeed) initialize(ctx cdcContext.Context) (err error) {
-	if c.initialized || c.state.Status == nil {
+func (c *changefeed) initialize(ctx cdcContext.Context,
+	cfInfo *model.ChangeFeedInfo,
+	cfStatus *model.ChangeFeedStatus,
+) (err error) {
+	if c.initialized || cfStatus == nil {
 		// If `c.state.Status` is nil it means the changefeed struct is just created, it needs to
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:458]: Please update the comments.

File: cdc/owner/owner.go [hunk 6]
```diff
@@ -173,13 +175,21 @@ func (o *ownerImpl) Tick(stdCtx context.Context, rawState orchestrator.ReactorSt
...
+			cfReactor = o.newChangefeed(changefeedID, changefeedState.Info, changefeedState.Status,
+				newFeedStateManager(up, changefeedState),
+				up, o.cfg)
 			o.changefeeds[changefeedID] = cfReactor
 		}
 		ctx = cdcContext.WithChangefeedVars(ctx, &cdcContext.ChangefeedVars{
 			ID: changefeedID,
 		})
-		cfReactor.Tick(ctx, o.getChangefeedCaptures(changefeedState, state))
+		changefeedState.CheckCaptureAlive(ctx.GlobalVars().CaptureInfo.ID)
```
Comment (asddongmen) [inline cdc/owner/owner.go:186]: Why this is needed?

File: cdc/owner/changefeed.go (hunk 1 above)
Comment (sdojjy) [reply cdc/owner/changefeed.go:52]: fixed

File: cdc/owner/changefeed.go (hunk 3 above)
Comment (sdojjy) [reply cdc/owner/changefeed.go]: done

File: cdc/owner/changefeed.go (hunk 5 above)
Comment (sdojjy) [reply cdc/owner/changefeed.go:455]: done

File: cdc/owner/owner.go (hunk 6 above)
Comment (sdojjy) [reply cdc/owner/owner.go:186]: just keep the old logic 
checks if the capture is alive, if the capture offline,  the etcd worker will exit and throw the ErrLeaseExpired error.

File: cdc/owner/changefeed.go (hunk 4 above)
Comment (sdojjy) [reply cdc/owner/changefeed.go:341]: done

File: cdc/owner/feed_state_manager.go [hunk 7]
```diff
@@ -94,24 +124,22 @@ func (m *feedStateManager) resetErrRetry() {
...
+			m.initCheckpointTs = m.state.Status.CheckpointTs
 		}
 	}
 
-	m.checkAndInitLastRetryCheckpointTs(state.Status)
+	if m.state != nil {
+		m.checkAndInitLastRetryCheckpointTs(m.state.Status)
+	}
 
-	m.state = state
```
Comment (CharlesCheung96) [inline cdc/owner/feed_state_manager.go:114]: Why remove this assginment statement?

**Review by CharlesCheung96 — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**